pkg/log/zap/flags.go

```diff
@@ -158,3 +168,44 @@ func (v sampleValue) IsBoolFlag() bool {
 func (v sampleValue) Type() string {
 	return "sample"
 }
+
+type timeformatValue struct {
```

Comment:
WDYT about making this `timeEncodingValue` and changing the flag name to `zap-time-encoding`? Basically use `encoding` instead of `format` everywhere since that's the terminology that zap is using.

Reply:
I agree - it makes more sense to keep things consistent with zap